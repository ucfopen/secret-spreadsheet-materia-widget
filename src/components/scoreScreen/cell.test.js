import React from 'react';
import Cell from './cell';
import renderer from 'react-test-renderer';

const props = () => {
	return {
		key: 1,
		id: 1,
		displayText: `Correct Answer`,
		answer: `Wrong Answer`,
		answerable: true,
		leftAlign: false,
		correct: false,
		checked: false
	};
};

const customProps = (changes={}) => {
	const newProps = props();

	Object.defineProperties(newProps, changes);

	return newProps;
};

describe(`Cell`, () => {
	beforeEach(() => {
		jest.resetModules();
		jest.resetAllMocks();
	});

	test(`Is rendered not answerable centered`, () => {
		const genProps = customProps({
			answerable: {value: false}
		});

		const component = renderer.create(<Cell {... genProps} />);
		const tree = component.toJSON();

		expect(tree).toMatchSnapshot();
	});

	test(`Is rendered not answerable left aligned`, () => {
		const genProps = customProps({
			answerable: {value: false},
			leftAlign: {value: true}
		});

		const component = renderer.create(<Cell {... genProps} />);
		const tree = component.toJSON();

		expect(tree).toMatchSnapshot();
	});

	test(`Is rendered answerable centered unchecked with correct response`, () => {
		const genProps = customProps({
			correct: {value: true},
			answer: {value: `Correct Answer`}
		});

		const component = renderer.create(<Cell {... genProps} />);
		const tree = component.toJSON();

		expect(tree).toMatchSnapshot();
	});

	test(`Is rendered answerable centered unchecked with wrong response`, () => {
		const genProps = props();

		const component = renderer.create(<Cell {... genProps} />);
		const tree = component.toJSON();

		expect(tree).toMatchSnapshot();
	});

	test(`Is rendered answerable centered checked with correct response`, () => {
		const genProps = customProps({
			correct: {value: true},
			answer: {value: `Correct Answer`},
			checked: {value: true}
		});

		const component = renderer.create(<Cell {... genProps} />);
		const tree = component.toJSON();

		expect(tree).toMatchSnapshot();
	});

	test(`Is rendered answerable centered checked with wrong response`, () => {
		const genProps = customProps({
			checked: {value: true}
		});

		const component = renderer.create(<Cell {... genProps} />);
		const tree = component.toJSON();

		expect(tree).toMatchSnapshot();
	});

	test(`Is rendered answerable left aligned unchecked with correct response`, () => {
		const genProps = customProps({
			leftAlign: {value: true},
			correct: {value: true},
			answer: {value: `Correct Answer`}
		});

		const component = renderer.create(<Cell {... genProps} />);
		const tree = component.toJSON();

		expect(tree).toMatchSnapshot();
	});

	test(`Is rendered answerable left aligned unchecked with wrong response`, () => {
		const genProps = customProps({
			leftAlign: {value: true},
		});

		const component = renderer.create(<Cell {... genProps} />);
		const tree = component.toJSON();

		expect(tree).toMatchSnapshot();
	});

	test(`Is rendered answerable left aligned checked with correct response`, () => {
		const genProps = customProps({
			leftAlign: {value: true},
			correct: {value: true},
			answer: {value: `Correct Answer`},
			checked: {value: true}
		});

		const component = renderer.create(<Cell {... genProps} />);
		const tree = component.toJSON();

		expect(tree).toMatchSnapshot();
	});

	test(`Is rendered answerable left aligned checked with wrong response`, () => {
		const genProps = customProps({
			leftAlign: {value: true},
			checked: {value: true}
		});

		const component = renderer.create(<Cell {... genProps} />);
		const tree = component.toJSON();

		expect(tree).toMatchSnapshot();
	});
});
