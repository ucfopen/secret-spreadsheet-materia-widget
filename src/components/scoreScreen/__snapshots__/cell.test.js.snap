// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cell Is rendered answerable centered checked with correct response 1`] = `
<td
  className="right"
  id={1}
>
  <p>
    Correct Answer
  </p>
</td>
`;

exports[`Cell Is rendered answerable centered checked with wrong response 1`] = `
<td
  className="right"
  id={1}
>
  <p>
    Correct Answer
  </p>
</td>
`;

exports[`Cell Is rendered answerable centered unchecked with correct response 1`] = `
<td
  className="right"
  id={1}
>
  <p>
    Correct Answer
  </p>
</td>
`;

exports[`Cell Is rendered answerable centered unchecked with wrong response 1`] = `
<td
  className="wrong"
  id={1}
>
  <p>
    Wrong Answer
  </p>
</td>
`;

exports[`Cell Is rendered answerable left aligned checked with correct response 1`] = `
<td
  className="right leftAlign"
  id={1}
>
  <p>
    Correct Answer
  </p>
</td>
`;

exports[`Cell Is rendered answerable left aligned checked with wrong response 1`] = `
<td
  className="right leftAlign"
  id={1}
>
  <p>
    Correct Answer
  </p>
</td>
`;

exports[`Cell Is rendered answerable left aligned unchecked with correct response 1`] = `
<td
  className="right leftAlign"
  id={1}
>
  <p>
    Correct Answer
  </p>
</td>
`;

exports[`Cell Is rendered answerable left aligned unchecked with wrong response 1`] = `
<td
  className="wrong leftAlign"
  id={1}
>
  <p>
    Wrong Answer
  </p>
</td>
`;

exports[`Cell Is rendered not answerable centered 1`] = `
<td
  className=""
  id={1}
>
  Correct Answer
</td>
`;

exports[`Cell Is rendered not answerable left aligned 1`] = `
<td
  className="leftAlign"
  id={1}
>
  Correct Answer
</td>
`;
